### ISSUE TITLE
Unclear documentation: line position off by 1 when using JSON formatter

### ISSUE THREAD

**Issue by theoretick** [palantir/tslint#4528]
### Bug Report

- __TSLint version__: 5.12.1
- __TypeScript version__: 3.3.3
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

Reproduction project: https://github.com/theoretick/tslint-position-bug-reproduction

```ts


// Non-literal require - tsr-detect-non-literal-require
var lib: String = 'fs'
require(lib)

// Eval with variable - tsr-detect-eval-with-expression
var myeval: String = 'console.log("Hello.");';
eval(myeval);

// Unsafe Regexp - tsr-detect-unsafe-regexp
const regex: RegExp = /(x+x+)+y/;

// Non-literal Regexp - tsr-detect-non-literal-regexp
var myregexpText: String = "/(x+x+)+y/";
var myregexp: RegExp = new RegExp(myregexpText);
myregexp.test("(x+x+)+y");

// Markup escaping disabled - tsr-detect-disable-mustache-escape
var template: Object = new Object;
template.escapeMarkup = false;

// Detects HTML injections - tsr-detect-html-injection
var element: Element =  document.getElementById("mydiv");
var content: String = "mycontent"
Element.innerHTML = content;

// Timing attack - tsr-detect-possible-timing-attacks
var userInput: String = "Jane";
var auth: String = "Jane";
if (userInput == auth) {
  console.log(userInput);
}
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "warning",
  "extends": [],
  "jsRules": {},
  "rules": {
    "no-eval": true
  },
  "rulesDirectory": []
}
```

#### Actual behavior

when using the default prose formatter, the line number is correctly reported. When I use the JSON formatter, however, the line reported is off by one (line listed as 7 not 8):

```
❯ npm run lint_prose

> tslint-line-position-reproduction@1.0.0 lint_prose /Users/theoretick/code/tmp/tslint-line-position-reproduction
> tslint -c tslint.json ./**/*.ts


WARNING: src/main.ts:8:1 - forbidden eval


❯ npm run lint_json

> tslint-line-position-reproduction@1.0.0 lint_json /Users/theoretick/code/tmp/tslint-line-position-reproduction
> tslint -t json -c tslint.json ./**/*.ts

[{"endPosition":{"character":4,"line":7,"position":201},"failure":"forbidden eval","name":"src/main.ts","ruleName":"no-eval","ruleSeverity":"WARNING","startPosition":{"character":0,"line":7,"position":197}}]
```

#### Expected behavior

When using `-t json` I would expect the `startPosition.line` to match the occurrence line in the prose formatter.

**Comment by JoshuaKGoldberg** [comment]
> the line reported is off by one (line listed as 7 not 8)

Hmm, that's an interesting discussion. The prose and stylish reporters are intended to be human readable, where we would consider the first line of a file to be 1. The JSON reporter is intended to be machine readable, where we could consider the first line of a file to be 0.

I'm tempted to close this as by-design. Thoughts?

**Comment by theoretick** [comment]
Haha, that didn't even occur to me but makes sense.  mm in that case, by-design makes sense.

It feels a bit non-intuitive to me but I suppose if I were using the prose linter I might assume it was zero-indexed already.

Perhaps it would be worth including a note regarding machine-readable formatters? https://palantir.github.io/tslint/formatters/

**Comment by JoshuaKGoldberg** [comment]
> Perhaps it would be worth including a note regarding machine-readable formatters? https://palantir.github.io/tslint/formatters/

Sorry to reopen after you already closed @theoretick, but I very much agree! This seems like a bug in the documentation. Accepting PRs to explicitly mention this.
